### DISCUSSION TITLE
Reparsing a node

### DISCUSSION THREAD

**Discussion by CAYdenberg** [remarkjs/remark#1155] [Q&A] [answered]
I'm a bit of a n00b to unified, but it seems like this should have a straightforward answer.

I am working with a flavour of markdown that (mis)uses alt text as captions. For example:

```
![**Bold title.** Rest of caption.](https://example.com/path/figure.jpg)
```

Once remarkParse has passed over it, the alt text is "Bold title. Rest of caption." with no rich text markup. Is there a standard approach (a util?) for going back to the original file so that I can select the text within the square brackets and reparse it as markdown?

**Comment by ChristianMurphy** [comment]
Welcome @CAYdenberg!
I'm not sure I follow your question.
Are you trying to go Markdown back to Markdown? Or Markdown to HTML, JSX, or React?

> remarkParse has passed over it, the alt text is "Bold title. Rest of caption." with no rich text markup

What are you looking at?
The AST? The virtual file? Or the contents written out at the end?

> back to the original file so that I can select the text within the square brackets and reparse it as markdown?

This sounds like your proposed solution, but what is your goal?
Are you trying to get remark to work as a drop in replacement for a different markdown renderer producing HTML? Are you looking to use remark to lint or format Markdown? Something else?

**Reply by CAYdenberg** [reply]
I am building a Slate-based editor for a markdown-like format. However, in existing documents which use this format, some people use rich text within the "square bracket" part of an image element. The rich text is ultimately rendered as a caption (not as alt text) in HTML or PDF.

The existing transforms are I believe based on Pandoc, though I would have to dig down and see how it's configured.

**Comment by wooorm** [comment, marked as answer]
> Once remarkParse has passed over it, the alt text is "Bold title. Rest of caption." with no rich text markup.

This is indeed the case. That is because CommonMark prescribes it as such.
This is because images and links are parsed the same: separately, `![` and `[`, and then `](url)` for example. Then when a `](url)` matches to a `![`, they together turn into an image.
That means they can include emphasis, strong, inline code, autolinks even, etcetera.
But of course: `<em>` doesn’t work in `alt="..."`. Which is why CommonMark prescribes that the HTML tags are ignored.

> Is there a standard approach 

No, I’d instead recommend alternatives to your custom markdown flavor which are more commonly supported, such as generic directives: https://github.com/remarkjs/remark-directive.

---

See also a related issue I posted on CM: https://github.com/commonmark/commonmark-spec/issues/716.

**Reply by CAYdenberg** [reply]
Thanks, that answers my question.
